lib/node_modules/@stdlib/array/base/map/lib/main.js

```diff
@@ -0,0 +1,161 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
+var toAccessorArray = require( '@stdlib/array/base/to-accessor-array' );
+var zeros = require( '@stdlib/array/base/zeros' );
+
+
+// FUNCTIONS //
+
+/**
+* Tests whether an object has a specified method.
+*
+* @private
+* @param {Object} obj - input object
+* @param {string} method - method name
+* @returns {boolean} boolean indicating whether an object has a specified method
+*
+* @example
+* var bool = hasMethod( [], 'map' );
+* // returns true
+*
+* @example
+* var bool = hasMethod( [], 'beep' );
+* // returns false
+*/
+function hasMethod( obj, method ) {
+	return ( typeof obj[ method ] === 'function' );
+}
+
+/**
+* Applies a callback function to elements in an input array and assigns results to elements in a new output array.
+*
+* @private
+* @param {Collection} x - input array
+* @param {Function} fcn - callback function
+* @param {*} thisArg - callback execution context
+* @returns {Collection} output array
+*
+* @example
+* var ones = require( '@stdlib/array/base/ones' );
+*
+* function scale( x ) {
+*     return x * 10.0;
+* }
+*
+* var x = ones( 4 );
+* var y = internal( x, scale );
+* // returns [ 10.0, 10.0, 10.0, 10.0 ]
+*/
+function internal( x, fcn, thisArg ) {
+	var out;
+	var i;
+
+	out = zeros( x.length );
+	for ( i = 0; i < x.length; i++ ) {
+		out[ i ] = fcn.call( thisArg, x[ i ], i, x );
+	}
+	return out;
+}
+
+/**
+* Applies a callback function to elements in an input array and assigns results to elements in a new output array.
+*
+* @private
+* @param {Object} x - input array object
+* @param {Function} fcn - callback function
+* @param {*} thisArg - callback execution context
+* @returns {Object} output array object
+*
+* @example
+* var toAccessorArray = require( '@stdlib/array/base/to-accessor-array' );
+* var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
+* var ones = require( '@stdlib/array/base/ones' );
+*
+* function scale( x ) {
+*     return x * 10.0;
+* }
+*
+* var x = arraylike2object( toAccessorArray( ones( 4 ) ) );
+* var y = accessors( x, scale );
+* // returns {...}
```

Comment:
Same comment.

Reply:
This sort of example does not allow our doctesting framework to serve as lightweight unit tests. Here, the returns annotation is not particularly informative.